doc/dependency-resolution.md

```diff
@@ -0,0 +1,98 @@
+## Dependency Resolution
+
+**Problem Statement**
```

Comment:
There's a much more basic one too, which is that all the links in the frontend metadata block of an item - eg organisation, policies, countries etc - are calculated at request time, which is unnecessary; we'd like to do it at publish time, but we need the link text to update automatically when the item's title itself changes.